mne/preprocessing/tests/test_maxwell.py

```diff
@@ -157,4 +163,66 @@ def test_maxwell_filter_additional():
     assert_equal(cov_raw_rank, raw.info['nchan'])
     assert_equal(cov_sss_rank, maxwell.get_num_moments(int_order, 0))
 
+
+@testing.requires_testing_data
+def test_maxwell_filter_fine_calibration():
+    """Test fine calibration feature of Maxwell filter
+
+    Notes
+    -----
+    Elekta MaxFilter uses single precision, so expect filtered results to
+    differ slightly.
+    """
+
+    # Load testing data (raw, SSS std origin, SSS non-standard origin)
+    with warnings.catch_warnings(record=True):  # maxshield
+        raw = Raw(raw_fname, preload=False, proj=False,
+                  allow_maxshield=True).crop(0., 1., False)
+    raw.preload_data()
+    raw.pick_types(meg=True, eeg=False)
+    with warnings.catch_warnings(record=True):  # maxshield, naming
+        sss_fine_cal = Raw(sss_fine_cal_fname, preload=True, proj=False,
+                           allow_maxshield=True)
+
+    # Create coils
+    all_coils, _, _, meg_info = _prep_meg_channels(raw.info, ignore_ref=True,
+                                                   elekta_defs=True)
+    picks = [raw.info['ch_names'].index(ch) for ch in [coil['chname']
+                                                       for coil in all_coils]]
+
+    # Test 1D SSS fine calibration
+    raw_sss = maxwell.maxwell_filter(raw, origin=[0., 0., 40.],
+                                     int_order=int_order, ext_order=ext_order,
+                                     fine_cal_fname=fine_cal_fname)
+    assert_array_almost_equal(raw_sss._data[picks, :],
+                              sss_fine_cal._data[picks, :], decimal=11,
+                              err_msg='Maxwell filtered data with fine '
+                              'calibration incorrect.')
+
+    # Confirm SNR is above 100
+    bench_rms = np.sqrt(np.mean(sss_fine_cal._data[picks, :] ** 2, axis=1))
+    error = raw_sss._data[picks, :] - sss_fine_cal._data[picks, :]
+    error_rms = np.sqrt(np.mean(error ** 2, axis=1))
+    assert_true(np.mean(bench_rms / error_rms) >= 100, 'SNR < 100')
+
+    # Test 3D SSS fine calibration
+    raw_sss = maxwell.maxwell_filter(raw, origin=[0., 0., 40.],
+                                     int_order=int_order, ext_order=ext_order,
+                                     fine_cal_fname=fine_cal_fname_3d)
+    assert_array_almost_equal(raw_sss._data[picks, :],
+                              sss_fine_cal._data[picks, :], decimal=11,
+                              err_msg='Maxwell filtered data with fine '
+                              'calibration incorrect.')
+
+    # Confirm SNR is above 100
+    bench_rms = np.sqrt(np.mean(sss_fine_cal._data[picks, :] ** 2, axis=1))
+    error = raw_sss._data[picks, :] - sss_fine_cal._data[picks, :]
+    error_rms = np.sqrt(np.mean(error ** 2, axis=1))
+    assert_true(np.mean(bench_rms / error_rms) >= 100, 'SNR < 100')
+
+
+# TODO: Eventually add simulation tests mirroring Taulu's original papers
```

Comment:
can you add a NIH Pubmed link here, or some other relevant link to the paper/papers?